include/sampleflow/consumers/autocovariance_dim_1.h

```diff
@@ -0,0 +1,255 @@
+         * matrix using this one sample.
+         *
+         * @param[in] sample The sample to process.
+         * @param[in] aux_data Auxiliary data about this sample. The current
+         *   class does not know what to do with any such data and consequently
+         *   simply ignores it.
+         */
+        virtual
+        void
+        consume (InputType     sample,
+                 AuxiliaryData aux_data) override;
+
+        /**
+         * A function that returns the covariance matrix computed from the
+         * samples seen so far. If no samples have been processed so far, then
+         * a default-constructed object of type InputType will be returned.
+         *
+         * @return The computed covariance matrix.
+         */
+        value_type
+        get () const;
+
+      private:
+        /**
+         * A mutex used to lock access to all member variables when running
+         * on multiple threads.
+         */
+
+        mutable std::mutex mutex;
+
+        /**
+         * The current value of $\bar x_k$ as described in the introduction
+         * of this class.
+         */
+
+        InputType           current_mean;
+
+        /**
+         * Currently describes lag of autocovariance
+         */
+
+        InputType 			k;
+
+        /**
+         * The current value of $\bar x_k$ as described in the introduction
+         * of this class.
+         */
+        boost::numeric::ublas::matrix<scalar_type> current_autocovariationdim1_matrix;
+
+        /**
+        * Save previous sample value
+        */
+
+        boost::numeric::ublas::matrix<scalar_type> past_sample;
+        boost::numeric::ublas::matrix<scalar_type> past_sample_replace;
+
+        /**
+         * The number of samples processed so far.
+         */
+        types::sample_index n_samples;
+    };
+
+    template <typename InputType>
+    autocovariancedim1 <InputType>::
+	autocovariancedim1  ()
+      :
+      n_samples (0)
+    {}
+
+
+    template <typename InputType>
+    void
+	autocovariancedim1 <InputType>::
+    consume (InputType sample, AuxiliaryData /*aux_data*/)
+    {
+      std::lock_guard<std::mutex> lock(mutex);
+
+      // If this is the first sample we see, initialize the matrix with
+      // this sample. After the first sample, the autocovariance matrix
+      // is the zero matrix since a single sample does not have any friends yet.
+
+      //Set how long is our autocovariance tail
+      k=10;
+
+      if (n_samples == 0)
+        {
+          n_samples = 1;
+          current_autocovariationdim1_matrix.resize (k,3);
+          for (unsigned int i=0; i<k; ++i)
+        	for (unsigned int j=0; j<2; ++j){
+        		current_autocovariationdim1_matrix(i,j) = 0;
+        	}
+          current_mean = sample;
+          past_sample.resize (k,1);
+          past_sample_replace.resize (k,1);
+          past_sample(0,0) = sample;
+        }
+      else if(n_samples < k)
+		{
+     	 ++n_samples;
```

Comment:
you ought to let eclipse indent the file for you so that things are nicely aligned :-)